test/test_helpers/twemproxy.rb

```diff
@@ -0,0 +1,168 @@
+require '3scale/backend/storage_async'
+require '3scale/backend/storage_sync'
+
+module TestHelpers
+  module Storage
+    # Test::Unit hooks, just include TestHelpers::Storage in a testcase if you
+    # do not use the global mocking with at_start/at_exit hooks.
+    def self.included(base)
+      base.singleton_class.instance_eval do
+        prepend Hooks
+      end
+    end
+
+    module Hooks
+      def startup
+        Mock.mock_storage_clients
+        super
+      end
+
+      def shutdown
+        super
+        Mock.unmock_storage_clients
+      end
+    end
+    private_constant :Hooks
+
+    module Mock
+      DEFAULT_TWEMPROXY_PORT = 22121
+
+      DEFAULT_NODES = %w[127.0.0.1:6382 127.0.0.1:6383 127.0.0.1:6384].freeze
+      private_constant :DEFAULT_NODES
+
+      STORAGE_CLASSES = [
+        ThreeScale::Backend::StorageAsync::Client,
+        ThreeScale::Backend::StorageSync
+      ].freeze
+      private_constant :STORAGE_CLASSES
+
+      class << self
+        def set_nodes(*nodes)
+          @nodes = nodes.flatten
+        end
+
+        def nodes
+          # Don't return non proxied nodes unless we are testing on a twemproxy
+          uri = URI(ThreeScale::Backend.configuration.redis.proxy)
+          return [uri.to_s] if uri.port != DEFAULT_TWEMPROXY_PORT
+
+          @nodes || DEFAULT_NODES
+        end
+
+        def mock_storage_clients
+          STORAGE_CLASSES.each { |klass| mock_storage_client!(klass) }
+        end
+
+        def unmock_storage_clients
+          STORAGE_CLASSES.each { |klass| unmock_storage_client!(klass) }
+        end
+
+        private
+
+        def mock_storage_client!(storage_client_class)
+          class << storage_client_class
+            # ensure this does not get overwritten
```

Comment:
Just wondering - where it can potentially be overwritten?

Reply:
No idea. I should have mentioned I copied and pasted this from the old code, when twemproxy was the only way to run tests: https://github.com/3scale/apisonator/pull/370/files#diff-51187addae661483cd1f02434a8e6f4ab93ebacf0b3fe7ae54e6949231696da5

There's a lot on this file that feels like arcane magic for me.